packages/pieces/src/lib/apps/hubspot/common/props.ts

```diff
@@ -0,0 +1,44 @@
+import { OAuth2PropertyValue, Property } from '../../../framework/property';
+import { hubSpotClient } from './client';
+
+export const hubSpotAuthentication = Property.OAuth2({
+  displayName: 'Authentication',
+  authUrl: 'https://app.hubspot.com/oauth/authorize',
+  tokenUrl: 'https://api.hubapi.com/oauth/v1/token',
+  required: true,
+  scope: [
+    'crm.lists.read',
+    'crm.lists.write',
+    'crm.objects.contacts.read',
+    'crm.objects.contacts.write',
+  ],
+});
+
+export const hubSpotListIdDropdown = Property.Dropdown<number>({
+  displayName: 'List',
+  refreshers: ['authentication'],
+  description: 'List to add contact to',
+  required: true,
+  options: async (propsValue) => {
+    if (propsValue['authentication'] === undefined) {
+      return {
+        disabled: true,
+        options: [],
+        placeholder: 'Please select an authentication',
+      }
+    }
+
+    const token = (propsValue['authentication'] as OAuth2PropertyValue).access_token;
+    const listsResponse = await hubSpotClient.lists.getStaticLists({ token });
+
+    const options = listsResponse.lists.map(list => ({
+      label: list.name,
+      value: list.listId,
+    }));
+
```

Comment:
Please add a check for when there are no lists and add a placeholder "Please create a list"

Reply:
updated in cb72ee8